src/abc_atlas_access/abc_atlas_cache/cloud_cache.py

```diff
@@ -1113,15 +1211,19 @@ def _download_file(self, file_attributes: CacheFileAttributes) -> bool:
             was_downloaded = True
             response = self.s3_client.get_object(Bucket=self._bucket_name,
                                                  Key=str(obj_key))
+            hasher = hashlib.md5()
 
             if 'Body' in response:
                 with open(local_path, 'wb') as out_file:
                     for chunk in response['Body'].iter_chunks():
                         out_file.write(chunk)
+                        if not skip_hash_check: hasher.update(chunk)
                         pbar.update(len(chunk))
 
-            # Verify the size of the downloaded file
-            if file_attributes.file_size != file_attributes.local_path.stat().st_size:
+            # Verify the hash of the downloaded file
+            test_checksum = hasher.hexdigest()
+            if not skip_hash_check and \
+                    test_checksum != file_attributes.file_hash:
                 file_attributes.local_path.exists()
```

Comment:
Someone should have caught this when I first submitted this line of code ~ 3 years ago. This `file_attributes.local_path.exists()` isn't doing anything. I think it was supposed to be a redundant

```
if file_attributes.local_path.exists():
```

Go ahead and delete this line.

Reply:
Is there a crash if you run `unlink` on a file that isn't there? I'll check.